### PR TITLE
cicd: add more coverage files, to display a more valid number

### PR DESCRIPTION
**Pull request by nandor-magyar**
We have not sent all the golang test results to **codecov**, addressing this in this PR.

### CODE REVIEW

File: .github/workflows/product_builder.yaml [hunk 1]
```diff
@@ -209,8 +216,35 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: golang-unit-coverage
+          path: ${{ env.GOLANG_WORKING_DIRECTORY }}/**.cov
+  go_coverage_upload:
+    runs-on: ubuntu-22.04
+    needs:
+      - go_security
+      - go_lint
+      - go_test
```
Comment (polaroi8d) [inline .github/workflows/product_builder.yaml:229]: If eg. `go_test` needs `gather_changes` is there any reason why we are adding in here? Or just a double check?

**Comment by polaroi8d** [comment]
Why is this branch showing the same percentage?

File: .github/workflows/product_builder.yaml (hunk 1 above)
Comment (nandor-magyar) [reply .github/workflows/product_builder.yaml:229]: No, I think it has no use. I'm removing it.

**Comment by nandor-magyar** [comment]
@polaroi8d the readme contains a link to an image of the develop branch.

**Comment by polaroi8d** [comment]
> @polaroi8d the readme contains a link to an image of the develop branch.

Ahh okey.

**Review by polaroi8d — APPROVED** [review]

**Pull request merged**